src/_pytest/config/__init__.py

```diff
@@ -1662,6 +1663,54 @@ def _warn_about_skipped_plugins(self) -> None:
             )
 
 
+class OutputVerbosity:
+    DEFAULT = "auto"
+    _option_name_fmt = "verbosity_{}"
+
+    def __init__(self, config: Config) -> None:
+        self._config = config
+
+    @property
+    def verbose(self) -> int:
+        """Application wide verbosity level."""
+        return cast(int, self._config.option.verbose)
+
+    def verbosity_for(self, output_type: str) -> int:
+        """Return verbosity level for the given output type.
+
+        :param output_type: Name of the output type.
+
+        If the level is not configured, the value of ``config.option.verbose``.
+        """
+        level = self._config.getini(OutputVerbosity._ini_name(output_type))
+
+        if level == OutputVerbosity.DEFAULT:
+            return self.verbose
+        return int(level)
+
+    @staticmethod
+    def _ini_name(output_type: str) -> str:
+        return f"verbosity_{output_type}"
+
+    @staticmethod
+    def add_ini(parser: "Parser", output_type: str, help: str) -> None:
```

Comment:
Since many things will add these type of settings, I thought it would be helpful to have this function to ensure it is dong consistently. I'm not completely in love with this level of indirection. But it is necessary since config settings are added to the `Parser` before the `Config` object exists.

Reply:
Good idea. 👍